cocos/ui/UIPageView.cpp

```diff
@@ -70,683 +62,192 @@ PageView* PageView::create()
     return nullptr;
 }
     
-void PageView::onEnter()
-{
-#if CC_ENABLE_SCRIPT_BINDING
-    if (_scriptType == kScriptTypeJavascript)
-    {
-        if (ScriptEngineManager::sendNodeEventToJSExtended(this, kNodeOnEnter))
-            return;
-    }
-#endif
-    
-    Layout::onEnter();
-    scheduleUpdate();
-}
-
 bool PageView::init()
 {
-    if (Layout::init())
+    if (ListView::init())
     {
-        setClippingEnabled(true);
+        setDirection(Direction::HORIZONTAL);
+        setMagneticType(MagneticType::CENTER);
+		setScrollBarEnabled(false);
         return true;
     }
     return false;
 }
 
-void PageView::addWidgetToPage(Widget *widget, ssize_t pageIdx, bool forceCreate)
+void PageView::setDirection(PageView::Direction direction)
 {
-    if (!widget || pageIdx < 0)
+    ListView::setDirection(direction);
+    if(direction == Direction::HORIZONTAL)
     {
-        return;
+        _indicatorPositionAsAnchorPoint = Vec2(0.5f, 0.1f);
     }
-   
-    ssize_t pageCount = this->getPageCount();
-    if (pageIdx < 0 || pageIdx >= pageCount)
+    else if(direction == Direction::VERTICAL)
     {
-        if (forceCreate)
-        {
-            if (pageIdx > pageCount)
-            {
-                CCLOG("pageIdx is %d, it will be added as page id [%d]",static_cast<int>(pageIdx),static_cast<int>(pageCount));
-            }
-            Layout* newPage = createPage();
-            newPage->addChild(widget);
-            addPage(newPage);
-        }
+        _indicatorPositionAsAnchorPoint = Vec2(0.1f, 0.5f);
     }
-    else
+
+    if(_indicator != nullptr)
     {
-        Node * page = _pages.at(pageIdx);
-        page->addChild(widget);
+        _indicator->setDirection(direction);
+        refreshIndicatorPosition();
     }
 }
 
-Layout* PageView::createPage()
+void PageView::addWidgetToPage(Widget *widget, ssize_t pageIdx, bool forceCreate)
```

Comment:
Here is a TAB, please change it to 4 spaces. And it happens many places in the header file and the implementation file.